app/src/main/java/comcmput301f16t01/github/carrier/ElasticRequestController.java

```diff
@@ -0,0 +1,78 @@
+package comcmput301f16t01.github.carrier;
+
+import android.os.AsyncTask;
+import android.util.Log;
+
+import com.searchly.jestdroid.DroidClientConfig;
+import com.searchly.jestdroid.JestClientFactory;
+import com.searchly.jestdroid.JestDroidClient;
+
+import java.io.IOException;
+
+import io.searchbox.core.DocumentResult;
+import io.searchbox.core.Index;
+import io.searchbox.core.Search;
+import io.searchbox.core.SearchResult;
+import io.searchbox.core.Update;
+
+/**
+ * Created by michael on 28/10/16.
+ */
+
+public class ElasticRequestController {
+    private static JestDroidClient client;
+
+    /**
+     * Called to cancel a request in elastic search
+     */
+    public static class CancelRequest extends AsyncTask<Request, Void, Void> {
+
+        /**
+         * Async task to update a request in elastic search to cancelled.
+         * @param requests
+         * @return
+         */
+        @Override
+        protected Void doInBackground(Request... requests) {
+            verifySettings();
+
+            for ( Request request : requests ) {
+
+                String script = "{\n" +
+                        "    \"script\" : \"ctx._source.status = " + Request.CANCELLED + "\" }";
```

Comment:
Pretty please comment things. I'm not all that familiar with elastic search, I can kind of tell what's going on here, but it would be a lot more clear with some comments.